src/gt4py/next/program_processors/runners/gtfn.py

```diff
@@ -141,11 +144,24 @@ def fingerprint_compilable_program(inp: stages.CompilableProgram) -> str:
 
 class FileCache(diskcache.Cache):
     """
-    This class extends `diskcache.Cache` to ensure the cache is closed upon deletion,
-    i.e. it ensures that any resources associated with the cache are properly
-    released when the instance is garbage collected.
+    This class extends `diskcache.Cache` to ensure the cache is properly
+    - opened on a distributed file system by using a file lock. This guards the creating of the
```

Comment:
Is writing not a problem? Except that it's much harder to make all processes write at the same time and there less likely to trigger.

Reply:
No, writing to the cache is actually fine. SQLite which is used internally by `diskcache.Cache` handles concurrency very well in most cases, except for changing settings of the database itself, and that is the problem here. On creation of the `diskcache.Cache` object, some sql `PRAGMA` statements are executed and they lead to the observed issues. It is not really clear why, this should work, but it doesn't.